NumLib/ODESolver/NonlinearSolver-impl.h

```diff
@@ -149,6 +165,10 @@ solve(Vector &x)
         }
     }
 
+    MathLib::GlobalMatrixProvider<Matrix, Vector>::provider.releaseMatrix(J);
```

Comment:
Every matrix or vector is released as soon as it is (temporarily) not used anymore. Therefore the calls happen where they naturally fit, and not in a single place. Or am I not getting your point?